ts/services/ElementService.ts

```diff
@@ -179,6 +157,118 @@ class ElementService {
     return response;
   }
 
+  public async elementSearch(config, logger): Promise<void> {
+
+    // base query
+    const requestBody = this.buildBaseElementQuery(config);
+
+    // add name term
+    if (typeof config.name !== 'undefined') {
+      const field = 'elementNames';
+      requestBody.body[field] = this.buildValueQueryTerm(config.name, true, false);
+    }
+    // add type term
+    if (typeof config.type !== 'undefined') {
+      const field = 'elementTypes';
+      requestBody.body[field] = this.buildValueQueryTerm(config.type, true, true);
+    }
+    // add attributes term
+    if (typeof config.attribute !== 'undefined') {
+      const attrKey = config.attribute.split('=')[0];
+      const attrVal = config.attribute.split('=')[1];
+
+      const field = 'attributes';
+      requestBody.body[field] = this.buildKeyValueQueryTerm(attrKey, attrVal);
+    }
+    // add elementTags term
+    if (typeof config.tag !== 'undefined') {
+      const tagKey = config.tag.split('=')[0];
+      const tagVal = config.tag.split('=')[1];
+
+      const field = 'elementTags';
+      requestBody.body[field] = this.buildKeyValueQueryTerm(tagKey, tagVal);
+    }
+    // add collectors term
+    if (typeof config.collector !== 'undefined') {
+      const field = 'collectorNames';
+      requestBody.body[field] = this.buildValueQueryTerm(config.collector, true, true);
+    }
+
+    logger.debug(JSON.stringify(requestBody, null, 2));
+
+    try {
+      const response = await request(requestBody);
+      if (config.format === 'text') {
+        logger.info(`###########################################################################`);
+        logger.info('## Element Search');
+        logger.info(`##  Page Size [${response.page.size}]`);
+        logger.info(`##  Page Number [${response.page.number} of ${response.page.totalPages - 1}]`);
+        logger.info(`##  Total Elements [${response.page.totalElements}]`);
+        logger.info(`###########################################################################`);
+        logger.info(response.page.content.map((el) => {
+          return `name[${el.name}] id[${el.id}] fqn[${el.fqn}] type[${el.type}]`;
```

Comment:
This output format was odd to read. What about a table?

Reply:
CSV?

Reply:
Still wouldn't be easy on the eyes. Maybe https://www.npmjs.com/package/table?